src/copilot/inspect/index.ts

```diff
@@ -1,21 +1,20 @@
 import { CancellationToken, CodeAction, CodeActionContext, CodeActionKind, ExtensionContext, TextDocument, languages, window, workspace, Range, Selection } from "vscode";
 import { COMMAND_INSPECT_RANGE, registerCommands } from "./commands";
-import { InspectActionCodeLensProvider } from "./InspectActionCodeLensProvider";
-import { DefaultRenderer as DefaultInspectionRenderer } from "./render/DefaultRenderer";
-import { InspectionRenderer } from "./render/InspectionRenderer";
+import { DocumentRenderer } from "./DocumentRenderer";
+import { fixDiagnostic } from "./render/DiagnosticRenderer";
 
-export async function activateCopilotInspection(context: ExtensionContext): Promise<void> {    
-    const inspectActionCodeLenses = new InspectActionCodeLensProvider().install(context);
-    const inspectionRenderer: InspectionRenderer = new DefaultInspectionRenderer().install(context);
-    // Commands
-    registerCommands(inspectionRenderer);
+export async function activateCopilotInspection(context: ExtensionContext): Promise<void> {
+    const renderer: DocumentRenderer = new DocumentRenderer().install(context);
+    registerCommands(renderer);
 
     context.subscriptions.push(
-        workspace.onDidOpenTextDocument(doc => inspectActionCodeLenses.rerender(doc)), // Rerender class codelens when open a new document
-        workspace.onDidChangeTextDocument(e => inspectActionCodeLenses.rerender(e.document)), // Rerender class codelens when change a document
-        languages.registerCodeActionsProvider({ language: 'java' }, { provideCodeActions: rewrite }), // add code action to rewrite code
+        languages.registerCodeActionsProvider({ language: 'java' }, { provideCodeActions: fixDiagnostic }), // Fix using Copilot
+        languages.registerCodeActionsProvider({ language: 'java' }, { provideCodeActions: rewrite }), // Inspect using Copilot
+        workspace.onDidOpenTextDocument(doc => renderer.rerender(doc)), // Rerender class codelens and cached suggestions on document open
+        workspace.onDidChangeTextDocument(e => renderer.rerender(e.document, true)), // Rerender class codelens and cached suggestions debouncely on document change
```

Comment:
When the text document is changed, shouldn't you re-inspect it again before re-render?

Reply:
`inspect` is expensive. Classes usually have multiple methods, if user modifies one method, inspections of other methods are still available (that is why we have cache and why we don't need to re-inspect the whole document again), but their positions may be changed (that's why we need to rerender).

Reply:
the question is if I introduce a new method to the file, when will it update the inspections?